src/setup_node_env/openssl_legacy_provider/index.js

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+var branch = require('../../../package.json').branch;
+var docsBranch = branch.match(/^\d\.\d\d?$/) || 'current';
+var openSSLLegacyProviderEnabled = require('./openssl_legacy_provider_enabled')();
+
+if (openSSLLegacyProviderEnabled) {
+  console.log(
```

Comment:
@elastic/kibana-core any thoughts on where I can put this check to hook into the core logger?  This is currently pre-startup, where we have similar node checks such as validating node.js's version, not running as root etc.  

The difference is this doesn't need to prevent startup, and the default experience with this log looks out of place.